example/src/domain/machinery/Form.js

```diff
@@ -34,6 +34,42 @@ export function FormCheckbox({ field, label }) {
   )
 }
 
+export function FormCheckboxGroupField({ field, options, label }) {
+  const { name, state, eventHandlers: { onChange, ...eventHandlers } } = useFormField(field)
+  const { value } = state
+
+  console.log(`[${name}] render checkbox group field`)
+
+  return (
+    <FieldsetAndError {...{ label, state }}>
+      {options.map((option, i) => (
+        <div key={`${name}__${i}`}>
+          <label htmlFor={name}>{option.label}</label>
+          <input
+            id={`${name}__${i}`}
+            type='checkbox'
+            value={option.value}
+            checked={Boolean(value) && value.includes(option.value)}
+            onChange={handleChange}
+            {...eventHandlers}
+            {...{ name }}
+          />
+        </div>
+      ))}
+    </FieldsetAndError>
+  )
+
+  function handleChange(e) {
+    const changedValue = e.currentTarget.value
+    const newValue =
+      !Array.isArray(value) ? [changedValue] : 
+      value.includes(changedValue) ? value.filter(x => x !== changedValue) :
+      value.concat(changedValue)
+
+    onChange(newValue)
+  }
+}
```

Comment:
Op deze manier zijn we niet afhankelijk van de 'toString' functie van de optie. Dus de value van de optie komt 1 op 1 terug in de `onChange` zonder dat die per ongeluk veranderd is doordat ie via `e.currentTarget.value` binnenkomt.

```suggestion
  function handleChange(changedValue) {
    return e => {
      const newValue =
        !Array.isArray(value) ? [changedValue] : 
        value.includes(changedValue) ? value.filter(x => x !== changedValue) :
        value.concat(changedValue)

      onChange(newValue)
    }
  }
}
```